### PR TITLE
NDI-2: Styled shoreditch for safe colours

### PR DESCRIPTION
**Pull request by mukeshcompucorp**
**Description:**
This PR contains major changes about new color scheme as per WCAG.
See new color palette and designs at: 
https://invis.io/7QCECB68Z#/254761833_WCAG_AA_Options
https://invis.io/7QCECB68Z#/256894652_Contact_Summary
https://invis.io/7QCECB68Z#/256894653_Personal_Details

**Screenshots:**

#### Before
https://user-images.githubusercontent.com/26058635/32374680-c4fea3ac-c0c4-11e7-975d-6dc42ceccc82.png

#### After
https://user-images.githubusercontent.com/26058635/32374399-80cebb50-c0c3-11e7-8e17-a8414a2398c5.png

#### Before
before-2

#### After
after-2

#### Before
before-3

#### After
after-3

#### Before
before-4

#### After
after-4

#### Before
before-5

#### After
after-5

#### Before
before-6

#### After
after-6

**Notes:**
This work should be tested extensively before merging with develop branch

- [x] Backstop tested

### CODE REVIEW

**Comment by mi-kon** [comment]
@mukeshcompucorp  please don't merge this until this is tested on CiviHR 
CC: @AkA84

**Comment by mi-kon** [comment]
@mukeshcompucorp please add at least one unique element screenshot before and after + add screenshot of design + change in styleguide

File: scss/bootstrap/components/_editable-content.scss [hunk 1]
```diff
@@ -2,6 +2,10 @@
   display: block;
   position: relative;
 
+  .crm-i {
+    opacity: 1;
```
Comment (AkA84) [inline scss/bootstrap/components/_editable-content.scss]: why?

File: scss/bootstrap/overrides/_variables.scss [hunk 2]
```diff
@@ -52,22 +52,17 @@ $gray-dark:                 $crm-copy !default;
 $gray-light:                $crm-background !default;
 $gray-lighter:              #F3F6F7 !default;
 $crm-gray-lightest:         #FAFAFB !default;
+$crm-black:                 #000000 !default;
```
Comment (AkA84) [inline scss/bootstrap/overrides/_variables.scss]: put this after `$crm-white`

File: scss/bootstrap/overrides/_variables.scss [hunk 3]
```diff
@@ -23,10 +23,10 @@ $crm-page-topbar-bg-light:     #393E46;
 $crm-page-topbar-color:        #fff !default;
 $crm-page-topbar-item-padding: 15px !default;
 $crm-accordion-header-color:   #f3f6f7;
-$crm-dasboard-header:          #5D677B;
+$crm-dasboard-header:          #4D4D69;
```
Comment (AkA84) [inline scss/bootstrap/overrides/_variables.scss]: 1) the variable has a typo
2) where is this used?
3) why are we defining a custom color for this instead of using a color from the palette?

File: scss/bootstrap/overrides/_variables.scss [hunk 4]
```diff
@@ -23,10 +23,10 @@ $crm-page-topbar-bg-light:     #393E46;
...
 $crm-h3-color:                 #525966;
 $crm-wizard-border:            #eff3f5;
-
+$crm-text-muted:               #9494A5;
```
Comment (AkA84) [inline scss/bootstrap/overrides/_variables.scss]: why is this necessary? seems yet another additional color

File: scss/bootstrap/overrides/_variables.scss [hunk 5]
```diff
@@ -52,22 +52,17 @@ $gray-dark:                 $crm-copy !default;
...
-$brand-primary-dark:        #005A97 !default;
-$brand-secondary-dark:      #3D3D54 !default;
-$brand-success-dark:        #2C8452 !default;
-$brand-info-dark:           #005A97 !default;
-$brand-warning-dark:        #9F6A26 !default;
-$brand-danger-dark:         #A52946 !default;
 
 $crm-help-success:          #d1e5ce;
-$crm-brand-primary-light:   lighten(desaturate(adjust-hue($brand-primary, 4.9301), 34.7499), 45.6863) !default;
+$crm-brand-primary-light:   #2574A9 !default;
```
Comment (AkA84) [inline scss/bootstrap/overrides/_variables.scss]: what is this used for? and why can't it be calculated as it was before?

File: scss/bootstrap/overrides/_variables.scss [hunk 6]
```diff
@@ -352,10 +347,21 @@ $state-info-text:                $brand-info !default;
 $state-warning-text:             $brand-warning !default;
 $state-danger-text:              $brand-danger !default;
 
-$alert-success-border:           $brand-success !default;
+$alert-success-border:           $btn-success-bg !default;
```
Comment (AkA84) [inline scss/bootstrap/overrides/_variables.scss]: wait, why are you using a button-related color (at least, that's what it looks like from the variable name) to define an alert related color? you should define generic color, not colors tied to other components

File: scss/bootstrap/overrides/_variables.scss [hunk 7]
```diff
@@ -352,10 +347,21 @@ $state-info-text:                $brand-info !default;
...
 $state-info-border:              $brand-info !default;
-$alert-warning-border:           $brand-warning !default;
+$alert-warning-border:           $btn-warning-bg !default;
 $alert-danger-border:            $brand-danger !default;
+
+$alert-success-bg:              lighten($btn-success-bg, 30%) !default;
```
Comment (AkA84) [inline scss/bootstrap/overrides/_variables.scss]: are these custom variables or are you overriding existing ones?

File: scss/bootstrap/overrides/style/_buttons.scss [hunk 8]
```diff
@@ -60,39 +60,39 @@
...
   }
 }
 
 .btn-success {
   color: $gray-darker;
 
   &:hover {
-    background-color: $brand-success-dark;
-    color: $crm-white;
+    background-color: darken($btn-success-bg, 7%);
```
Comment (AkA84) [inline scss/bootstrap/overrides/style/_buttons.scss]: this '7%' looks to me that should be in some variable, it's everywhere

File: scss/civicrm/common/_tables.scss [hunk 9]
```diff
@@ -315,6 +315,25 @@ table {
...
+.dataTables_wrapper {
+  .dataTables_paginate {
+    .paginate_button {
+      color: $gray-darker !important;
+
+      &:hover:not(.disabled),
+      &.current {
+        background: none;
+        border: none;
+        font-weight: 600;
```
Comment (AkA84) [inline scss/civicrm/common/_tables.scss]: isn't there a variable for this value

File: scss/bootstrap/components/_editable-content.scss (hunk 1 above)
Comment (mukeshcompucorp) [reply scss/bootstrap/components/_editable-content.scss]: @AkA84 this was making color less visible (80%) of it's original color which should not be the case.

File: scss/bootstrap/overrides/_variables.scss (hunk 4 above)
Comment (mukeshcompucorp) [reply scss/bootstrap/overrides/_variables.scss]: This is from palette, we have it for disabled text

File: scss/bootstrap/overrides/_variables.scss (hunk 7 above)
Comment (mukeshcompucorp) [reply scss/bootstrap/overrides/_variables.scss]: This is overriding of existing variables.

File: scss/bootstrap/overrides/style/_buttons.scss (hunk 8 above)
Comment (mukeshcompucorp) [reply scss/bootstrap/overrides/style/_buttons.scss]: Ok I will make related changes

File: scss/bootstrap/overrides/_variables.scss (hunk 6 above)
Comment (mukeshcompucorp) [reply scss/bootstrap/overrides/_variables.scss]: @AkA84 for alert type warning and success, we have slightly different accent then original branding.
Now, in default bootstrap as well, we have this variables overridden as $brand-success while we should have it for this light accent. Should I keep it as this or should I change it adapt value #44CB7E

Also advise for https://github.com/mukeshcompucorp/org.civicrm.shoreditch/blob/fd677954177f9e3bd28b2181e9edd5d325bda6dc/scss/bootstrap/overrides/_variables.scss#L81-L92

File: scss/bootstrap/components/_editable-content.scss (hunk 1 above)
Comment (AkA84) [reply scss/bootstrap/components/_editable-content.scss]: "this" what? what was making the opacity different?

File: scss/bootstrap/overrides/_variables.scss (hunk 4 above)
Comment (AkA84) [reply scss/bootstrap/overrides/_variables.scss]: then definite first as a color `$crm-some-color` and then assign it to the text-muted (where did you get the name?) variable

File: scss/bootstrap/components/_editable-content.scss (hunk 1 above)
Comment (mukeshcompucorp) [reply scss/bootstrap/components/_editable-content.scss]: Default CiviCRM style.

File: scss/bootstrap/overrides/_variables.scss (hunk 4 above)
Comment (mukeshcompucorp) [reply scss/bootstrap/overrides/_variables.scss]: ok, I created name $crm-text-muted my self so, it one can recognize it as muted/disabled text

File: scss/bootstrap/components/_datepicker.scss [hunk 10]
```diff
@@ -9,7 +9,6 @@
         }
 
         tbody {
-            background-color: $crm-brand-primary-light;
```
Comment (AkA84) [inline scss/bootstrap/components/_datepicker.scss]: why is this not needed anymore?

File: scss/bootstrap/components/_datepicker.scss [hunk 11]
```diff
@@ -24,10 +23,6 @@
                     outline: 0 none;
                 }
 
-                &:hover {
-                    background: $crm-brand-primary-lighter;
```
Comment (AkA84) [inline scss/bootstrap/components/_datepicker.scss]: why is this not needed anymore?

File: scss/bootstrap/overrides/_variables.scss [hunk 12]
```diff
@@ -22,11 +22,8 @@ $crm-page-topbar-bg:           #222831 !default;
 $crm-page-topbar-bg-light:     #393E46;
 $crm-page-topbar-color:        #fff !default;
 $crm-page-topbar-item-padding: 15px !default;
-$crm-accordion-header-color:   #f3f6f7;
-$crm-dasboard-header:          #4D4D69;
-$crm-h3-color:                 #525966;
-$crm-wizard-border:            #eff3f5;
-$crm-text-muted:               #9494A5;
```
Comment (AkA84) [inline scss/bootstrap/overrides/_variables.scss]: none of this variables are needed anymore?

File: scss/bootstrap/overrides/_variables.scss [hunk 13]
```diff
@@ -22,11 +22,8 @@ $crm-page-topbar-bg:           #222831 !default;
...
+
+$crm-placeholder:              #9494A5;
```
Comment (AkA84) [inline scss/bootstrap/overrides/_variables.scss]: is this color part of the palette? Also, can you please give it a name like `$crm-{name_of_color}` which you then assign to `crm-placeholder`?

btw, are we sure about this name? Why can't we just use `$crm-{name_of_color}`? What if we want to use this color with something that is not a placeholder?

File: scss/bootstrap/overrides/_variables.scss [hunk 14]
```diff
@@ -52,51 +49,34 @@ $gray-dark:                 $crm-copy !default;
...
 
 $brand-primary:             #0071BD !default;
-$brand-success:             #4D994D !default;
+$brand-success:             #44CB7E !default;
 $brand-info:                #0071BD !default;
-$brand-warning:             #BF5900 !default;
+$brand-warning:             #E6AB5E !default;
 $brand-danger:              #CF3458 !default;
 
+$crm-darken-percentage:     7% !default;
```
Comment (AkA84) [inline scss/bootstrap/overrides/_variables.scss]: don't put this in the middle of color definition, put it at the end

Comment (AkA84) [inline scss/bootstrap/overrides/_variables.scss]: also, we don't need a lighten-percentage?

File: scss/bootstrap/overrides/_variables.scss [hunk 15]
```diff
@@ -52,51 +49,34 @@ $gray-dark:                 $crm-copy !default;
...
-$crm-brand-primary-lighter: lighten(desaturate(adjust-hue(#3ab1c9, -10.0699), 36.9721), 48.2353) !default;
 
-$crm-color-neutral:         #949caa !default; // gray shade
 $crm-dark-blue:             #586277 !default;
 $crm-dark-grayblue:         #D3DEE2 !default;
 $crm-darker-grayblue:       #C2CFD8 !default;
-$crm-white:                 #FFF !default;
+$crm-white:                 #FFFFFF !default;
+$crm-black:                 #000000 !default;
+$crm-color-neutral:         $crm-darker-grayblue !default;
```
Comment (AkA84) [inline scss/bootstrap/overrides/_variables.scss]: if `$crm-darker-grayblue` is already defined, why do we need `color-neutral`?

File: scss/bootstrap/overrides/_variables.scss [hunk 16]
```diff
@@ -52,51 +49,34 @@ $gray-dark:                 $crm-copy !default;
...
 
-$crm-dialog-container-border: #d3dee2;
+$crm-accordion-header-color:$gray-lighter;
```
Comment (AkA84) [inline scss/bootstrap/overrides/_variables.scss]: missing whitespace

File: scss/bootstrap/overrides/_variables.scss [hunk 17]
```diff
@@ -139,33 +119,29 @@ $icon-font-path: "../vendor/bootstrap/fonts/bootstrap/" !default;
 //
 //## Common layout spacings, margins, and paddings
 
-$crm-gap-small: 5px !default;
+$crm-gap-small:               5px !default;
```
Comment (AkA84) [inline scss/bootstrap/overrides/_variables.scss]: no need for all that whitespace

File: scss/bootstrap/overrides/_variables.scss [hunk 18]
```diff
@@ -326,67 +297,68 @@ $crm-nav-tabs-panel-heading-link-weight:             600 !default;
...
-$pagination-disabled-bg:               $pagination-bg !default;
+$pagination-disabled-color:      $gray-light !default;
+$pagination-disabled-bg:         $pagination-bg !default;
 
 //== Form states and alerts
 //
 //## Define colors for form feedback states and, by default, alerts.
 
-$state-success-text:             $brand-success !default;
+$state-success-text:             #4D994D !default;
```
Comment (AkA84) [inline scss/bootstrap/overrides/_variables.scss]: where is this color coming from? is it part of the palette? if no, it should be removed. if yes, it should be declared first of all as a primary color, then assigned to this variable

File: scss/bootstrap/overrides/_variables.scss [hunk 19]
```diff
@@ -326,67 +297,68 @@ $crm-nav-tabs-panel-heading-link-weight:             600 !default;
...
 $state-info-text:                $brand-info !default;
-$state-warning-text:             $brand-warning !default;
+$state-warning-text:             #BF5900 !default;
```
Comment (AkA84) [inline scss/bootstrap/overrides/_variables.scss]: where is this color coming from? is it part of the palette? if no, it should be removed. if yes, it should be declared first of all as a primary color, then assigned to this variable

File: scss/bootstrap/overrides/style/_ui-bootstrap.scss [hunk 20]
```diff
@@ -11,7 +11,6 @@
             }
 
             tbody {
-                background-color: $crm-brand-primary-light;
```
Comment (AkA84) [inline scss/bootstrap/overrides/style/_ui-bootstrap.scss]: we don't need this anymore?

File: scss/bootstrap/overrides/style/_ui-bootstrap.scss [hunk 21]
```diff
@@ -26,10 +25,6 @@
                         outline: 0 none;
                     }
 
-                    &:hover {
-                        background: $crm-brand-primary-lighter;
```
Comment (AkA84) [inline scss/bootstrap/overrides/style/_ui-bootstrap.scss]: we don't need this anymore?

File: scss/civicrm/common/_tables.scss [hunk 22]
```diff
@@ -95,14 +95,6 @@ table {
     &.label {
       vertical-align: middle !important;
     }
-
-    .crm-table-group-summary td {
-      background: $crm-dark-blue !important;
-
-      &:last-child {
-        text-align: left;
```
Comment (AkA84) [inline scss/civicrm/common/_tables.scss]: we don't need this anymore?

File: scss/bootstrap/components/_datepicker.scss (hunk 10 above)
Comment (mukeshcompucorp) [reply scss/bootstrap/components/_datepicker.scss]: `.dropdown-menu[datepicker-popup-wrap]` actually check the wrapper of this. I've checked all angluar datepicker as well and this is no where used. also, this file is out dated as angular datepicker now uses `uib-datepicker-popup-wrap`

I have removed this line as I do not want to increase more scope this this particular ticket by bumping in such changes

File: scss/bootstrap/components/_datepicker.scss (hunk 11 above)
Comment (mukeshcompucorp) [reply scss/bootstrap/components/_datepicker.scss]: same as above

File: scss/bootstrap/overrides/_variables.scss (hunk 12 above)
Comment (mukeshcompucorp) [reply scss/bootstrap/overrides/_variables.scss]: no it should be one of the palette variable actually as defined in our palette.

File: scss/bootstrap/overrides/_variables.scss (hunk 15 above)
Comment (mukeshcompucorp) [reply scss/bootstrap/overrides/_variables.scss]: `$crm-color-neutral` is variable used in one of the component of datepicker.
The variable is going to get overriden in that component but, to prevent potential break in that until we decide where a component should be added. 
We need to have this color to allow other extensions to override it.
https://github.com/compucorp/civihr-employee-portal-theme/blob/staging/civihr_default_theme/scss/base/_variables.scss#L29

File: scss/bootstrap/overrides/_variables.scss (hunk 17 above)
Comment (mukeshcompucorp) [reply scss/bootstrap/overrides/_variables.scss]: This is added to keep file look good only.

File: scss/bootstrap/overrides/_variables.scss (hunk 18 above)
Comment (mukeshcompucorp) [reply scss/bootstrap/overrides/_variables.scss]: This is from our palette,
For Success and Warning we have non-primary color for text and icons which in bootstrap is handled by `$state-success-text` that's why it is in here without adding new variable.

File: scss/bootstrap/overrides/_variables.scss (hunk 19 above)
Comment (mukeshcompucorp) [reply scss/bootstrap/overrides/_variables.scss]: same as above

File: scss/bootstrap/overrides/style/_ui-bootstrap.scss (hunk 20 above)
Comment (mukeshcompucorp) [reply scss/bootstrap/overrides/style/_ui-bootstrap.scss]: This is not affecting component design. it's tested

File: scss/bootstrap/overrides/style/_ui-bootstrap.scss (hunk 21 above)
Comment (mukeshcompucorp) [reply scss/bootstrap/overrides/style/_ui-bootstrap.scss]: This is not affecting component design. it's tested

File: scss/civicrm/common/_tables.scss (hunk 22 above)
Comment (mukeshcompucorp) [reply scss/civicrm/common/_tables.scss]: We have moved style related to summery tooltip to common and this is just duplication of it which is not longer required.

File: scss/bootstrap/overrides/_variables.scss (hunk 18 above)
Comment (AkA84) [reply scss/bootstrap/overrides/_variables.scss]: @mukeshcompucorp any color in the palette should be defined in a "color variable" first, so we can have all the colors bundled together in the same area of the variables file. Then you assign the variable to the component variable. Ideally there should be no hex value defined in the whole variable file except for that "color area"

File: scss/bootstrap/overrides/_variables.scss (hunk 19 above)
Comment (AkA84) [reply scss/bootstrap/overrides/_variables.scss]: see above
